### PR TITLE
#184948247 Configure ESLint and Git Pre-commit hook

### PR DESCRIPTION
**Pull request by wilbrord2**
What does this PR do?
This PR setup ESLint and Git Pre-commit hook

Description of Task to be completed?

- installing eslint 
- configure eslint for react 
- setting react rules

#### How should this be manually tested?
`npm run format`

What are the relevant pivotal tracker stories?
Story ID: [#184948247](https://www.pivotaltracker.com/story/show/184948247)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      "ecmaVersion": "latest",
+      "sourceType": "module"
+    },
+    "settings": {
+      "react": {
+        "version": "detect"
+      }
+    },
+    "plugins": ["react"]
+  }
```
Comment (Emile-SHUMBUSHO) [inline .eslintrc.json:35]: Remove an empty line from your codes

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+"singleQuote": true,
+"trailingComma": "all",
+"jsxSingleQuote": false,
+"useTabs": true,
+"tabWidth": 4,
+"ignorePatterns": ["node_modules/", "dist/"],
+"parser": "typescript",
+"disableLanguages": ["html", "markdown"],
+"proseWrap": "always"
+}
```
Comment (Emile-SHUMBUSHO) [inline .prettierrc.json:12]: Remove an empty line from your codes

File: package.json [hunk 3]
```diff
@@ -9,18 +9,32 @@
     "build": "webpack --mode production",
     "start": "npm run build && serve -s dist",
     "storybook": "storybook dev -p 6006",
-    "build-storybook": "storybook build"
+    "build-storybook": "storybook build",
+    "format": "eslint --fix src/**/*.js"
```
Comment (Emile-SHUMBUSHO) [inline package.json:24]: Tried to test this on my local computer, but didn't find any find any results
Screenshot 2023-04-19 at 21 20 45

File: package.json [hunk 4]
```diff
@@ -9,18 +9,32 @@
...
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "src/**/*.js": [
+      "npm run format",
+      "npm run test"
```
Comment (Emile-SHUMBUSHO) [inline package.json:34]: using this `npm run test ` isn't going to bring a conflict with other test, such unit testing or they will work together? need more clarifications.

**Review by Emile-SHUMBUSHO — CHANGES_REQUESTED** [review]

File: .prettierrc.json (hunk 2 above)
Comment (wilbrord2) [reply .prettierrc.json:12]: Done

File: package.json (hunk 3 above)
Comment (wilbrord2) [reply package.json:24]: Try to declare a variable and don't use it you will get errors

Comment (wilbrord2) [reply package.json:24]: This is one if examples

File: package.json (hunk 4 above)
Comment (wilbrord2) [reply package.json:34]: This mean that you can not run test as long as you still have errors and warning in your code 
So it Will not cause any conflicts

**Review by Emile-SHUMBUSHO — APPROVED** [review]

**Review by AbdulKhaliq59 — APPROVED** [review]

**Review by khevin2 — APPROVED** [review]

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**